src/main/kotlin/xyz/crossward/controller/GroupController.kt

```diff
@@ -0,0 +1,50 @@
+package xyz.crossward.controller
+
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.transaction.annotation.Transactional
+import org.springframework.web.bind.annotation.*
+import xyz.crossward.entities.Group
+import xyz.crossward.service.GroupService
+import java.net.URLDecoder
+
+@RestController
+@RequestMapping("/api")
+class GroupController(
+        var service: GroupService
+) {
+
+    @PostMapping("/groups/telegram")
+    @ResponseStatus(HttpStatus.CREATED)
+    fun createTelegramGroup(@RequestBody group: Group): Group {
+        return service.createTelegramGroup(group)
+    }
+
+    @PostMapping("/groups/website")
+    @ResponseStatus(HttpStatus.CREATED)
+    fun createGroup(@RequestBody group: Group): Group {
+        return service.createWebsiteGroup(group)
+    }
+
+    @GetMapping("/groups/ids/{id}")
+    @ResponseStatus(HttpStatus.OK)
+    @Transactional(readOnly = true)
+    fun getGroupById(@PathVariable("id") id: String): ResponseEntity<Group> {
+
+        return ResponseEntity.ok(service.findGroupById(id))
+    }
+
+
+    @GetMapping("/groups/names/{name}")
```

Comment:
Wondering if we need {name} or if we can just do /groups/names like in UserController. Not sure what the difference is

Reply:
Yeah we should change this from a path variable to a query parameter. Look at the previous comment for details.